docs/hive-integration/baremetal-agent-controller.md

```diff
@@ -202,6 +202,23 @@ used to tell the installer what disk to use as the installation disk. Refer to t
 
 BMAC is always setting `automatedCleaningMode: disabled` even if the `BareMetalHost` manifest specifies another value (e.g. `automatedCleaningMode: metadata`). This may be changed in the future releases, but currently we do not support using Ironic to clean the node.
 
+---
+**NOTE**
+
+`RootDeviceHints` from the `BareMetalHost` resource are passed down to the `Agent` and from there handled by the service. If the hints provided to the BMH do not match any drive discovered in the inspection phase, BMAC sets `agent.spec.installation_disk_id: /dev/not-found-by-hints`. This will cause the service to notice that the specified disk does not exists and fail the respective validation, showing the following condition in the `Agent` resource
```

Comment:
```suggestion
BMAC uses the `RootDeviceHints` from the `BareMetalHost` resource to find a matching disk in the corresponding `Agent`'s inventory, and then sets that disk in `agent.Spec.InstallationDiskID`. If the hints provided to the BMH do not match any disk discovered in the discovery phase, BMAC sets the disk in the Agent's Spec as `/dev/not-found-by-hints`. This will cause the service to notice that the specified disk does not exist and fail the respective validation, showing the following condition in the `Agent` resource.
```